core/host_api/impl/crypto_extension.cpp

```diff
@@ -721,4 +730,16 @@ namespace kagome::host_api {
     SL_TRACE_FUNC_CALL(logger_, res, signature, msg, pubkey);
     return res;
   }
+
+  void CryptoExtension::reset() {
+    batch_verify_.reset();
+  }
+
+  runtime::WasmSize CryptoExtension::batchVerify(runtime::WasmSize ok) {
```

Comment:
Should we keep verifying signatures in a batch once one verification failed?

Reply:
Also, we should probably use 
https://docs.rs/schnorrkel/latest/schnorrkel/fn.verify_batch.html
Here for performance reasons.

Reply:
Each call to `* _batch_verify` must return separate result, so we must verify each signature and can't use batch

Reply:
According to the spec, _batch_verify returns 1 if the signature was successfully batched, not if it was successfully verified (but it can also return 1 for successful verification if batching was not enabled). So they can be batched and then processed all at once.

Reply:
It's okay if you don't want to bother with it in this PR, it's more of an optimization AFAIK, just leave a TODO issue if so.

Reply:
[rust](https://github.com/paritytech/polkadot-sdk/blob/ccd55552a85ae45227abf7e0a4aee803b707229c/substrate/primitives/io/src/lib.rs#L891-L904) returns `1` only if signature is verified